Description: Rename static library from libmariadbclient.a to libmariadb.a
 The upstream build system creates libmariadb.so.2, but the static library
 is libmariadbclient.a -- which incidentally conflicts with the GPL client
 library included with MariaDB.  This fixes the static library naming to be
 consistent.  Also, for the purposes of submitting to upstream, on Windows
 systems where having the same name would create conflicts between the DLL
 stub library and the static library, it renames the static library to
 libmariadb_static.lib.
Author: Daniel Schepler <schepler@debian.org>
Forwarded: https://lists.launchpad.net/maria-developers/msg07924.html

Index: mariadb-client-lgpl/libmariadb/CMakeLists.txt
===================================================================
--- mariadb-client-lgpl.orig/libmariadb/CMakeLists.txt
+++ mariadb-client-lgpl/libmariadb/CMakeLists.txt
@@ -404,6 +404,14 @@ SET_TARGET_PROPERTIES(libmariadb PROPERT
    ${CPACK_PACKAGE_VERSION_MAJOR}
    SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
 
+IF(WIN32)
+  SET_TARGET_PROPERTIES(mariadbclient PROPERTIES
+    OUTPUT_NAME libmariadb_static)
+ELSE()
+  SET_TARGET_PROPERTIES(mariadbclient PROPERTIES
+    OUTPUT_NAME mariadb)
+ENDIF()
+
 #
 # Installation
 #
@@ -417,8 +425,8 @@ IF(NOT WIN32 AND WITH_MYSQLCOMPAT)
   SET(INSTALL_PATH ${LIB_INSTALL_DIR}/${LIBSUFFIX_INSTALL_DIR})
   create_symlink(libmysqlclient${CMAKE_SHARED_LIBRARY_SUFFIX} libmariadb ${INSTALL_PATH})
   create_symlink(libmysqlclient_r${CMAKE_SHARED_LIBRARY_SUFFIX} libmariadb ${INSTALL_PATH})
-  create_symlink(libmysqlclient${CMAKE_STATIC_LIBRARY_SUFFIX} mariadbclient ${INSTALL_PATH})
-  create_symlink(libmysqlclient_r${CMAKE_STATIC_LIBRARY_SUFFIX} mariadbclient ${INSTALL_PATH})
+  create_symlink(libmysqlclient${CMAKE_STATIC_LIBRARY_SUFFIX} libmariadb ${INSTALL_PATH})
+  create_symlink(libmysqlclient_r${CMAKE_STATIC_LIBRARY_SUFFIX} libmariadb ${INSTALL_PATH})
 ENDIF()
 
 
